tests/perf/perf_tests.cc

```diff
@@ -203,6 +204,119 @@ static inline std::ostream& operator<<(std::ostream& os, duration d)
     return os;
 }
 
+/**
+ * A column object encapsulates the logic needed to print one
+ * type of result value, usually as a column (or a json attribute).
+ *
+ * This allows all printers to share a common view of the available
+ * columns.
+ */
+struct column {
+    static constexpr int default_width = 11;
+
+    using print_fn = std::function<void(const column&, std::FILE *file, const result& r)>;
+
+    template <typename F>
+    column(sstring header, int prec, F fn) : header{header}, prec{prec} {
+        using result_t = std::invoke_result_t<F,const result&>;
+        constexpr auto is_integral = std::is_integral_v<result_t>;
+        constexpr auto is_double = std::is_same_v<result_t, double>;
+        constexpr auto is_duration = std::is_same_v<result_t, duration>;
+        if constexpr (is_integral || is_double || is_duration) {
+            static constexpr std::string_view fmt_str = is_double ? "{:>{}.{}f}": "{:>{}}";
+            text_printer = [=](const column& c, std::FILE *file, const result& r) {
+                fmt::print(file, fmt_str, fn(r), c.width, c.prec);
+            };
+        } else {
+            static_assert(false, "unsupported return type");
+        }
```

Comment:
instead of having this `static_assert()` in the `else` branch, IMHO, would be simper to put 
```c++
static_assert(is_integral || is_double || is_duration, "unsupported return type");
# initiailize text_printer and double_fn
```

what do you think?